src/engine/sparqlExpressions/SparqlExpressionGenerators.h

```diff
@@ -127,25 +134,16 @@ inline auto valueGetterGenerator =
 /// Do the following `numItems` times: Obtain the next elements e_1, ..., e_n
 /// from the `generators` and yield `function(e_1, ..., e_n)`, also as a
 /// generator.
-inline auto applyFunction = [](auto&& function, size_t numItems,
-                               auto... generators)
-    -> cppcoro::generator<std::invoke_result_t<
-        decltype(function),
-        ql::ranges::range_value_t<decltype(generators)>...>> {
-  // A tuple holding one iterator to each of the generators.
-  std::tuple iterators{generators.begin()...};
-
-  auto functionOnIterators = [&function](auto&&... iterators) {
-    return function(AD_MOVE(*iterators)...);
-  };
-
-  for (size_t i = 0; i < numItems; ++i) {
-    co_yield std::apply(functionOnIterators, iterators);
-
-    // Increase all the iterators.
-    std::apply([](auto&&... its) { (..., ++its); }, iterators);
-  }
-};
+inline auto applyFunction =
+    [](auto&& function, [[maybe_unused]] size_t numItems, auto... generators) {
+      return ::ranges::views::zip(ad_utility::RvalueView{
+                 ad_utility::OwningView{std::move(generators)}}...)
+             // TODO<joka921> currently not moving the function.
+             | ::ranges::views::transform(
```

Comment:
Also I assume that in this case you could use `ql::ranges` but for consistency with the upper variant where it can't be used, you don't right?